### ISSUE TITLE
Szybkie poprawki jakości treningu — 5 konkretnych problemów

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#26]
## Przegląd

Analiza kodu ujawniła 5 konkretnych problemów obniżających jakość treningu. Pierwsze 4 można naprawić jedną zmianą w `config.toml`.

---

## 1. Learning rate zbyt niski — `config.toml:42`

**Problem:** `lr = 0.00001` (1e-5) to wartość 10x za niska dla tej architektury (Transformer + Conv1D + Dueling DQN). Powoduje ekstremalnie wolną konwergencję. Gradient clipping na poziomie 1.0 (`trainer.py:471`) zabezpiecza przed destabilizacją przy wyższym LR.

**Fix:**
```toml
lr = 0.0001
```

---

## 2. Reward clipping zbyt agresywny — `config.toml:89`

**Problem:** `clip_max = 2.0` przy prowizji ~0.65% oznacza, że transakcje o zysku 2%, 3%, 5% dostają **identyczną nagrodę**. Model nie potrafi rozróżnić dobrej transakcji od znakomitej — sygnał jest skompresowany.

**Fix:**
```toml
clip_max = 5.0
```

---

## 3. Epsilon decay liczy od złego punktu — `trainer.py:243-247`

**Problem:** Epsilon zaczyna opadać od kroku 0, ale trening startuje dopiero po zapełnieniu bufora (`min_buffer_size = 50_000`). Przy pierwszym gradiencie epsilon jest już w połowie drogi — wczesna eksploracja jest zmarnowana.

**Obecny kod:**
```python
total_decay_steps = self.epsilon_decay_fraction * self.config['training']['buffer_capacity']
self.epsilon = max(
    self.epsilon_end,
    self.epsilon_start - (self.epsilon_start - self.epsilon_end) * (self.step_count / max(total_decay_steps, 1))
)
```

**Fix:**
```python
total_decay_steps = self.epsilon_decay_fraction * self.config['training']['buffer_capacity']
training_steps = max(0, self.step_count - self.min_buffer_size)
self.epsilon = max(
    self.epsilon_end,
    self.epsilon_start - (self.epsilon_start - self.epsilon_end) * (training_steps / max(total_decay_steps, 1))
)
```

Alternatywnie jako szybkie obejście bez zmiany kodu:
```toml
epsilon_decay_fraction = 0.5   # było: 0.1
```

---

## 4. Batch size zbyt duży — `config.toml:43`

**Problem:** `batch_size = 2048` przy `buffer_capacity = 1_000_000` oznacza, że każde doświadczenie jest próbkowane średnio ~24 razy zanim zostanie nadpisane. Duże batche zwiększają koszt obliczeniowy bez proporcjonalnego zysku — przy tej wielkości sieci 512 daje lepszy stosunek szybkości iteracji do jakości gradientu.

**Fix:**
```toml
batch_size = 512
```

---

## 5. Sparse rewards — brak nagrody pośredniej — `node/env/reward.js`

**Problem:** Nagroda wynosi 0 przez cały czas trwania pozycji. Sygnał pojawia się **tylko przy zamknięciu**. Przez 100+ kroków HOLDa sieć nie dostaje żadnego gradientu. Co więcej, `intermediate_reward_max = 0.1` jest już zdefiniowany w `config.toml`, ale **nigdzie nie jest używany**.

**Fix:** Dodać małą nagrodę z niezrealizowanego P&L podczas HOLDa:
```javascript
// w tradingEnv.step(), gdy pozycja jest otwarta ale nie zamykana
function getIntermediateReward(position, currentPrice, rewardConfig) {
    if (!position) return 0;
    const pnl = position.side === 'LONG'
        ? (currentPrice - position.openPrice) / position.openPrice
        : (position.openPrice - currentPrice) / position.openPrice;
    const scaled = pnl * rewardConfig.intermediate_reward_max;
    return Math.max(-0.1, Math.min(0.1, scaled));
}
```

---

## Podsumowanie — ranking według wpływu

| # | Problem | Plik | Wpływ | Trudność |
|---|---------|------|-------|----------|
| 1 | Learning rate zbyt niski | `config.toml:42` | Wysoki | Trivial |
| 2 | Reward clipping zbyt agresywny | `config.toml:89` | Wysoki | Trivial |
| 3 | Epsilon decay od złego punktu | `trainer.py:243` | Średni | Łatwy |
| 4 | Batch size zbyt duży | `config.toml:43` | Średni | Trivial |
| 5 | Brak nagrody pośredniej | `reward.js` | Średni | Średni |

**Natychmiastowy quick-win** (tylko `config.toml`):
```toml
lr = 0.0001
clip_max = 5.0
batch_size = 512
epsilon_decay_fraction = 0.5
```
